### PR TITLE
Utils for statvar processor for string lookup

### PR DESCRIPTION
**Pull request by ajaits**
Utils for statvar processor:

- ngram_matcher: partial substring lookup
- property_value_mapper: map from input string to DC property:values
- property_value_cache: cache of property:value hits/miss

### CODE REVIEW

File: tools/statvar_importer/mcf_file_util.py [hunk 1]
```diff
@@ -353,8 +409,10 @@ def load_mcf_nodes(
         ...
       }
   """
+    if nodes is None:
+      nodes = {}
```
Comment (rohitkumarbhagat) [inline tools/statvar_importer/mcf_file_util.py:413]: Usually we should avoid reassigning method arguments, as references are passed by value and reassigning wont make it visible outside the method. Can we create a local reference (like local_nodes = nodes?nodes:{}) and work with it within the method?

File: tools/statvar_importer/mcf_file_util.py [hunk 2]
```diff
@@ -213,21 +213,31 @@ def add_pv_to_node(
...
-            return node
-    if not value:
-        return node
+              return node
+    # allow empty values
+    # if not value:
+    #    return node
     existing_value = node.get(prop)
-    if existing_value and prop != 'Node' and prop != 'dcid':
+    if existing_value is not None and prop != 'Node' and prop != 'dcid':
```
[the remark this answers is not in the document]
Comment (ajaits) [reply tools/statvar_importer/mcf_file_util.py:237]: Some of statvar processor configs set value of a property to empty string to skip the property. These are later dropped when writing to the file.

File: tools/statvar_importer/mcf_file_util.py [hunk 3]
```diff
@@ -147,15 +147,15 @@ def strip_namespace(value: str) -> str:
...
-    Args:
-      value: string to be cleaned.
+  Args:
+    value: string to be cleaned.
 
-    Returns:
-      string without extra leading and trailing spaces.
-    """
+  Returns:
+    string without extra leading and trailing spaces.
```
[the remark this answers is not in the document]
Comment (ajaits) [reply tools/statvar_importer/mcf_file_util.py:154]: Added note on strip within double quotes.

File: tools/statvar_importer/mcf_file_util.py [hunk 4]
```diff
@@ -147,15 +147,15 @@ def strip_namespace(value: str) -> str:
...
+  """
     if value and isinstance(value, str):
```
[the remark this answers is not in the document]
Comment (ajaits) [reply tools/statvar_importer/mcf_file_util.py:156]: isinstance() is a py builtin that checks the type of the object

File: tools/statvar_importer/mcf_file_util.py (hunk 1 above)
Comment (ajaits) [reply tools/statvar_importer/mcf_file_util.py:413]: python seems to always pass by reference and changes are visible outside the function.
The caller is expected to get a dict object, even if empty.

**Pull request merged**